### ISSUE TITLE
Synchronising the X-axis range of two charts and allowing for synchronised zooming

### ISSUE THREAD

**Issue by qasimwarraich** [leeoniya/uPlot#816]
Firstly, thanks for all the hard work on this library and for being so responsive in the issues' tracker :). (it encouraged me to open an issue).

I have more of a question rather than an issue. I am dealing with two stacked charts that I have in cursor sync with each other. The X-axis data for both charts is in the form of timestamps, but one chart may have more data points than the other. All the time stamps are coerced to the same time of day, which enables the charts to represent "daily" values. The issue I am facing is that while I can get the charts to sync while zooming, the initial render ("un-zoomed state") creates different ranges for both charts. I am trying to figure out a way for the charts to have a shared range (the bigger of the two time ranges) so that the data points for the respective days on both charts line up visually.

I have tried writing a custom ranging function but kept running into edge cases which either meant having one side of the chart be mismatched or to not allow for zooming. Furthermore, I have also tried to utilise the  `u.status` property, which seemed initially promising, but it doesn't work when one zooms back out.

Basically, I am trying to conform the temporal ranges of two synced charts to be the same in the fully zoomed out state. I hope I was able to explain my question clearly enough. I would be incredibly grateful for some insight or an exchange on potential approaches.

If there is any more explanation or detail I can provide please let me know, and I thank you very much in advance :)

**Comment by leeoniya** [comment]
maybe something like this?

https://jsfiddle.net/f5jzd9qk/1/

```js
let data0 = [
  [0, 1, 2, 3],
  [5, 0, 5, 9],
];

let data1 = [
  [0, 1, 2, 3, 4, 5, 6],
  [1, 2, 3, 0, 3, 2, 1],
];

let minX = Math.min(data0[0][0], data1[0][0]);
let maxX = Math.max(data0[0].at(-1), data1[0].at(-1));

const outerLimits = (u, from, to) => {
  if (from === u.data[0][0] && to === u.data[0].at(-1)) {
    return [minX, maxX];
  }

  return [from, to];
};

const opts = {
  width: 800,
  height: 300,
  cursor: {
    sync: {
      key: 1,
    }
  },
  scales: {
    x: {
      time: false,
      range: outerLimits,
    },
  },
  series: [
    {},
    {
      stroke: null,
    },
  ],
};

opts.series[1].stroke = "red";
let u0 = new uPlot(opts, data0, document.body);
opts.series[1].stroke = "blue";
let u1 = new uPlot(opts, data1, document.body);
```
